apps/back-end/src/routes.ts

```diff
@@ -0,0 +1,147 @@
+
+  if (!fs.existsSync(file)) return false;
+
+  const stream = fs.createReadStream(file, "utf8");
+  reply.header("Content-Type", "application/json");
+  reply.send(stream);
+
+  return true;
+}
+
+//////////////////////////////////////////////////////////////////////
+
+/**
+ * Constructor for the MykoMap API router implementation.
+ *
+ * Accepts the router configuration, and returns an implementation using that configuration.
+ *
+ * See the Contract definition for details and documentation of the routes.
+ *
+ * Implementation Note: this is not a class, as ts-rest will attempt to recurse over its
+ * members, and any non-router members seem to make it explode when it does that.
+ * Therefore, we use a closure to store the configuration options, out of harm's way.
+ */
+export function MykomapRouter(
+  opts: MykomapRouterConfig,
+): RouterImplementation<Contract> {
+  // Validation
+  if (opts?.mykomap?.dataRoot == undefined)
+    // deliberately loose check
+    throw new Error("mandatory dataRoot option is not defined");
+
+  if (!fs.existsSync(opts.mykomap.dataRoot))
+    throw new Error(
+      `the dataRoot plugin option is set but refers to a non-existing path: `+
+        `'${opts.mykomap.dataRoot}'.`,
+    );
+
+  // Concatenates the path components into an absolute file path
+  const filePath = (...components: string[]): string => {
+    const p = path.join(opts.mykomap.dataRoot ?? "", ...components) + ".json";
+    return p;
+  };
+
+  // Construct and return the implementation object
+  return {
+    async getDataset({ params: { datasetId }, request, reply }) {
+      // Validate the parameters some more
+
+      if (!sendJson(request, reply, filePath("datasets", datasetId)))
+        throw new TsRestResponseError(contract.getDataset, {
+          status: 404,
+          body: { message: `unknown datasetId '${datasetId}'` },
+        });
+
+      return reply;
+    },
+
+    async searchDataset({
+      params: { datasetId },
+      query: { filter, text },
+      request,
+      reply,
+    }) {
+      const filter2 = filter ?? [];
+
+      const components = ["datasets", datasetId, "search", ...filter2, "text"];
+      if (text !== undefined) components.push(encodeURIComponent(text));
+
+      if (!sendJson(request, reply, filePath(...components)))
+        return {
+          status: 200,
+          body: [],
+        };
+
+      return reply;
+    },
+
+    async getDatasetItem({
+      params: { datasetId, datasetItemId },
+      request,
+      reply,
+    }) {
+      if (
+        !sendJson(
+          request,
+          reply,
+          filePath("datasets", datasetId, "items", String(datasetItemId)),
+        )
+      )
+        throw new TsRestResponseError(contract.getDatasetItem, {
+          status: 404,
+          body: { message: `item retrieve failed` },
+        });
+
+      return reply;
+    },
+
+    async getVersion(req) {
+      return {
+        body: __BUILD_INFO__,
+        status: 201,
```

Comment:
Should this be 200?

Reply:
I think this is a hang-over from the swagger-generated version wanting to use 201 for successful responses ("created successfully") - in this case (and the other cases in the API)  nothing is created, so I agree 200 is better.